File: Ruby/problem7.rb
def nth_prime(n)
  primes_found = 0
  candidate = 0
  while(primes_found < n) do
    candidate += 1
    primes_found += 1 if prime?(candidate)
  end
  candidate
end

def approximate_nth_prime(n)
  (n * Math.log(n)).to_i
end

def prime?(n)
  max = Math.sqrt(n).ceil
  for i in 2..max
    return false if n % i === 0
  end
  true
end

n = ARGV.first ? ARGV.first.to_i : 10001
start_time = Time.now
puts "Pn approximately is: #{approximate_nth_prime(n)}"
puts "#{Time.now - start_time} seconds."
start_time = Time.now
puts "Pn is: #{nth_prime(n)}"
puts "#{Time.now - start_time} seconds."

File: C#/problem7.cs
using System;
using System.Diagnostics;

class Problem7
{
  public static void Main(string[] args)
  {
    int n = 10001;
    if (args.Length > 0)
    {
      n = Convert.ToInt32(args[0]);
    }
    Stopwatch stopwatch = new Stopwatch();
    stopwatch.Start();
    Console.WriteLine("Approximately: " + ApproximateNthPrime(n));
    stopwatch.Stop();
    Console.WriteLine(stopwatch.ElapsedMilliseconds / 1000.0 + " seconds");

    stopwatch.Start();
    Console.WriteLine("Exactly: " + NthPrime(n));
    stopwatch.Stop();
    Console.WriteLine(stopwatch.ElapsedMilliseconds / 1000.0 + " seconds");

  }

  public static int NthPrime(int n)
  {
    int primesFound = 0;
    int candidate = 0;
    while (primesFound < n)
    {
      candidate++;
      if (Prime(candidate))
      {
        primesFound++;
      }
    }

    return candidate;
  }

  public static int ApproximateNthPrime(int n)
  {
    int approximate = Convert.ToInt32(n * System.Math.Log(n));
    return approximate;
  }

  public static bool Prime(int n)
  {
    long max = Convert.ToInt64(Math.Ceiling(Math.Sqrt(n)));
    bool prime = true;
    for (long i = 2; i <= max; i++)
    {
      if (n % i == 0)
      {
        prime = false;
        i = max;
      }
    }
    return prime;
  }
}
